### PR TITLE
Feat: Remove experimental decorators check from SWC config

### PR DESCRIPTION
**Pull request by nonzzz**
# Background

#40

## Checklists

- [x] Detect typescript version.
- [x] Add shims for `esm` bundle.
- [x] Add unit test for decorators.

### CODE REVIEW

**Comment by nonzzz** [comment]
BTW. I think variable `enaleExperimentalDecorators` should be rename?

File: src/index.ts [hunk 1]
```diff
@@ -107,6 +113,23 @@ function swc(options: PluginOptions = {}): RollupPlugin {
           ? {}
           : getOptions(this, dirname(id), options.tsconfig);
 
+      let enaleExperimentalDecorators = false;
+      if (isTypeScript) {
+        try {
+          const tsPath = require.resolve('typescript', { paths: [process.cwd()] });
+          const packageJsonPath = lookup(tsPath, 'package.json');
+          const { version } = JSON.parse(fs.readFileSync(packageJsonPath, 'utf-8'));
```
Comment (SukkaW) [inline src/index.ts]: If you are using `typescript` anyway, why not `require('typescript/package.json')`?

Comment (nonzzz) [reply src/index.ts]: emmm. I'll fixed it.

File: src/index.ts [hunk 2]
```diff
@@ -107,6 +106,22 @@ function swc(options: PluginOptions = {}): RollupPlugin {
           ? {}
           : getOptions(this, dirname(id), options.tsconfig);
 
+      let enaleExperimentalDecorators = false;
```
Comment (pionxzh) [inline src/index.ts]: -> `enableExperimentalDecorators` ?

File: src/index.ts [hunk 3]
```diff
@@ -107,6 +106,22 @@ function swc(options: PluginOptions = {}): RollupPlugin {
...
+      if (isTypeScript) {
+        try {
+          const packageJsonPath = require.resolve('typescript/package.json', { paths: [process.cwd()] });
+          const { version } = JSON.parse(fs.readFileSync(packageJsonPath, 'utf-8'));
+          const [major] = version.split('.');
+          // typescript 5.x
+          if (+major >= 5) enaleExperimentalDecorators = true;
+        } catch {
+          this.warn({
+            message: 'Can\'t find typescript, please Check typescript has been installed.',
```
Comment (pionxzh) [inline src/index.ts]:
```suggestion
            message: 'Failed to find TypeScript. Please check if TypeScript has been installed.',
```

File: src/index.ts (hunk 2 above)
Comment (nonzzz) [reply src/index.ts]: I noticed it. But i don't know what is better :(

Comment (pionxzh) [reply src/index.ts]: I mean, it's a typo ((

File: src/index.ts [hunk 4]
```diff
@@ -107,6 +106,22 @@ function swc(options: PluginOptions = {}): RollupPlugin {
...
+          const [major] = version.split('.');
+          // typescript 5.x
+          if (+major >= 5) enableExperimentalDecorators = true;
+        } catch {
+          this.warn({
+            message: 'Failed to find TypeScript. Please check if TypeScript has been installed.',
+            pluginCode: 'SWC_TYPESCRIPT_NOT_EXISTS'
+          });
+        }
+      }
```
Comment (SukkaW) [inline src/index.ts]: Should we run this during every `transform` phase?

IMHO we should hoist this to the plugin initialization.

**Review by SukkaW — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 4 above)
Comment (nonzzz) [reply src/index.ts]: I modified it.

**Pull request closed without merging**